File: content/kb/scala/limitar-tiempo-vida/src/password.scala
import scala.collection.JavaConversions._

object Password {

  def main(args: Array[String]): Unit = {
    val c: Console = System.console()
    if (c == null) {
      System.err.println("No console.")
      System.exit(1)
    }
    val username: String = c.readLine("Enter your user name: ")
    val password: Array[Char] = c.readPassword("Enter your password: ")
    val isValidUser: Boolean = verify(username, password)
    // limpiar la contraseña
    Arrays.fill(password, ' ')
    if (!isValidUser) {
      throw new SecurityException("Invalid Credentials")
    }
  }

  // Validación simulada, siempre devuelve true
  private def verify(username: String, password: Array[Char]): Boolean = true

}

File: content/defends/csharp/limitar-tiempo-vida/src/password.cs
using System;
using System.Security;

class Password
{
	static void Main(string[] args)
	{
		Console.WriteLine("Enter your user name: ");
		var loginid = Console.ReadLine();
		Console.WriteLine("Enter your password: ");
		var password = Console.ReadLine();
		char[] mArray = password.ToCharArray();

		bool isValidUser = verify(loginid, mArray);

		// limpiar la contraseña
		MyArray.Fill(mArray,' ');

		if (!isValidUser)
		{
			throw new SecurityException("Invalid Credentials");
		}
	}

	// Validación simulada, siempre devuelve true
	private static bool verify(string username, char[] password)
	{
		return true;
	}
}

internal static class MyArray
{

	internal static void Fill<T>(T[] array, T value)
	{
		for (int i = 0; i < array.Length; i++)
		{
			array[i] = value;
		}
	}

}